src/osx/Installer.Mac/uninstall.sh

```diff
@@ -1,22 +1,23 @@
 #!/bin/bash
 
-# Unconfigure GCM
-HELPER="$(git config --system credential.helper)"
-if [ "$HELPER" = "/usr/local/share/gcm-core/git-credential-manager-core" ]
+# Ensure we're running as root
+if [ $(id -u) != "0" ]
 then
-	echo "Resetting credential helper to 'osxkeychain'..."
-	sudo git config --system credential.helper osxkeychain
-else
-	echo "GCM was not configured as the Git credential helper."
+	sudo "$0" "$@"
+	exit $?
```

Comment:
we try to sudo and exit?

Reply:
Yes.. if we are not running as root we invoke ourselves `"@0"` with all args `"$@"` with sudo to force running as root. We then exit the script returning the last error code `$?`.